utils/navigator_tools/nodes/tf_fudger.py

```diff
@@ -52,18 +53,22 @@
 toTfAng = lambda x: x * ang_res - 0.5 * ang_range
 
 p = q = None
+q_mode = False
 
-try:
-    while not rospy.is_shutdown():
-        x, y, z = toTfLin(cv2.getTrackbarPos("x", "tf")), toTfLin(cv2.getTrackbarPos("y", "tf")), toTfLin(cv2.getTrackbarPos("z", "tf"))
-        p = (x, y, z)
-        rpy = (toTfAng(cv2.getTrackbarPos("roll", "tf")), toTfAng(cv2.getTrackbarPos("pitch", "tf")), toTfAng(cv2.getTrackbarPos("yaw", "tf")))
-        q = tf.transformations.quaternion_from_euler(*rpy)
+while not rospy.is_shutdown():
+    x, y, z = toTfLin(cv2.getTrackbarPos("x", "tf")), toTfLin(cv2.getTrackbarPos("y", "tf")), toTfLin(cv2.getTrackbarPos("z", "tf"))
+    p = (x, y, z)
+    rpy = (toTfAng(cv2.getTrackbarPos("roll", "tf")), toTfAng(cv2.getTrackbarPos("pitch", "tf")), toTfAng(cv2.getTrackbarPos("yaw", "tf")))
+    q = tf.transformations.quaternion_from_euler(*rpy)
 
-        print "xyz:", [round(x, 5) for x in p], "  rpy:", [round(np.degrees(x), 5) for x in rpy]
-        br.sendTransform(p, q, rospy.Time.now(), args.tf_child, args.tf_parent)
+    rpy_feedback = "xyz: {}, euler: {}".format([round(x, 5) for x in p], [round(np.degrees(x), 5) for x in rpy])
+    q_feedback = "xyz: {},     q: {}".format([round(x, 5) for x in p], [round(x, 5) for x in q])
+    print q_feedback if q_mode else rpy_feedback
 
-        cv2.waitKey(10)
-except KeyboardInterrupt:
-    print  "Final TF:", p, q
-    cv2.destroyAllWindows()
\ No newline at end of file
+    k = cv2.waitKey(10) & 0xFF
+    if k == ord('q'):
+        q_mode = not q_mode
+
+    br.sendTransform(p, q, rospy.Time.now(), args.tf_child, args.tf_parent)
+
+print  "\nFinal TF:", p, q
```

Comment:
It would be really nice if this could output a line for the new static_transform publisher that you could just copy and paste into the tf launch file.